free-1.md

```diff
@@ -57,12 +116,37 @@ making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+I'd say maybe 20-30% of their time would be spent on engaging in the language aspects. Some questions that remain to be answered:
+
+* Should code structures be opt-in or opt-out in the flowchart?
+..* Based on this decision, what does it look like to opt in or out, say, an `if` statement?
+* What language should be used to generate the flowchart?
+..* What effect do potential languages have on the processing logic?
+
 
 ### Scope
 _How big an idea is this? How ambitious is this project?_
 
+I think the idea has potential to grow, but I think one could successfully see multiple meaningful iterations
+before the semester's end.
+
 
 ### Benefits and drawbacks
 _Why might this be a good idea for a project? Why might this not be a good idea 
 project?_
 
+**Benefits**
+* Well-scoped
+* Have to consider adapting the proposed DSL to the language for which it's designed (considering both this language's syntax and style)
+* Have to consider complexity of flowchart creation language
+* Potential to extend or modify an existing language
+
+**Drawbacks**
+* There is an existing, though incomplete, language that addresses the need fairly well.
+
+
+[Javadoc]: https://en.wikipedia.org/wiki/Javadoc
+[Doxygen]: https://en.wikipedia.org/wiki/Doxygen
+[code2flow]: http://code2flow.com/
+[Flowgen]: http://jlopezvi.github.io/Flowgen/index.html
+[Flowgen Paper]: http://arxiv.org/pdf/1405.3240.pdf
```

Comment:
I like this idea. Let me see if I understand it correctly: A program in your DSL shows up as syntax in a general-purpose language, e.g., as comments and annotations. The runtime for your language then compiles the program into a flowchart. Is that correct?

I think this idea could make for a very good project. I would be a little concerned that you would spend too much time at first on things unrelated to language design, e.g., making the flowcharts look nice or trying to extract extra information by analyzing the general-purpose program. To counteract this temptation, I recommend:
   a. Choosing a general-purpose language that has a good parser API: a library that allows you access to the AST (or other intermediate representation) so that you don't have to worry about parsing the general-purpose language. That way, you can focus on parsing your DSL, if needed.
   b. Compiling (i.e., translating) your language into an existing DSL that can describe flowcharts (e.g., one of the [GraphViz](http://www.graphviz.org/) languages).